### PR TITLE
Companion Fixes (#6592)

### LINKED ISSUE #6592: Companion 2.3 issues
EEprom (xlite):

[eeprom-2000-01-30-085816.zip](https://github.com/opentx/opentx/files/3439423/eeprom-2000-01-30-085816.zip)

Issues : RSSI critical display 42 instead of 33
Hardware switch name for SC split in 2 (Ac and t)
RSSI source need to be chhecked too

### PR DESCRIPTION
**Pull request by raphaelcoeffic**
fixes #6592 

TODO:
- [x] fix offset in switch names.
- [x] load/save RSSI alarms
- [x] implement support for RSSI source.
- [x] fix special functions (see eeprom in #6592)
- [x] rename FrskyField to Frsky218Field and VariaFlield (without telem screens).

### CODE REVIEW

File: companion/src/firmwares/opentx/opentxeeprom.cpp [hunk 1]
```diff
@@ -42,7 +42,18 @@ inline int MAX_SWITCHES(Board::Type board, int version)
   return Boards::getCapability(board, Board::Switches);
 }
 
-#define MAX_SWITCH_SLOTS(board, version)      (IS_TARANIS_X9E(board) ? 32 : (version >= 219 ? 16 : 8))  // bitsize of swconfig_t / 2 (see radio/src/datastructs.h)
+// bitsize of swconfig_t / 2 (see radio/src/datastructs.h)
```
Comment (bsongis) [inline companion/src/firmwares/opentx/opentxeeprom.cpp:45]: yes, one less, thanks!

File: companion/src/firmwares/opentx/opentxeeprom.cpp [hunk 2]
```diff
@@ -1717,18 +1728,16 @@ class TelemetryCurrentSourceConversionTable: public ConversionTable
     }
 };
 
-class FrskyField: public StructField {
+class VarioField: public StructField {
```
Comment (bsongis) [inline companion/src/firmwares/opentx/opentxeeprom.cpp]: I am not sure it is easier to read now, as we mix telemetry screens (218) in a structure which should deal with vario.
Instead we could create a class Frsky218Field and another class VarioField (without screens inside).

Comment (raphaelcoeffic) [reply companion/src/firmwares/opentx/opentxeeprom.cpp]: ok, will do!

File: companion/src/firmwares/opentx/opentxeeprom.cpp [hunk 3]
```diff
@@ -1766,6 +1777,20 @@ class FrskyField: public StructField {
     TelemetryCurrentSourceConversionTable telemetryCurrentSourceConversionTable;
 };
 
+class VarioField: public StructField {
+  public:
+    VarioField(DataField * parent, FrSkyData & frsky, Board::Type board, unsigned int version, unsigned int variant):
```
Comment (bsongis) [inline companion/src/firmwares/opentx/opentxeeprom.cpp:1782]: the vario struct could even been splitted from FrSkyData, no?

File: companion/src/firmwares/opentx/opentxeeprom.cpp [hunk 4]
```diff
@@ -2018,20 +2043,23 @@ OpenTxModelData::OpenTxModelData(ModelData & modelData, Board::Type board, unsig
     }
   }
 
-  // TODO rename to VARIO
-  internalField.Append(new FrskyField(this, modelData.frsky, modelData.rssiAlarms, board, version, variant));
-
-  if (version >= 219) {
-    // TODO RSSI SOURCE
-    internalField.Append(new SpareBitsField<8>(this));
+  if (version <= 218) {
```
Comment (bsongis) [inline companion/src/firmwares/opentx/opentxeeprom.cpp:2046]: yes, perfect

**Comment by bsongis** [comment]
I merge it, the VarioData split is something different which could be done later

**Pull request merged**